core/src/test/java/org/mule/tck/testmodels/mule/TestMessageDispatcher.java

```diff
@@ -73,4 +95,10 @@ protected void doStop()
     {
         // no op
     }
+
+    @Override
+    protected boolean isSupportsNonBlocking()
```

Comment:
I spent a while moving to the approach you suggest but it just complicates functional testing, and doesn't add much value and there are no existing tests that depend on this.  We currently also use a single dispatcher for testing success/failure caes, rather than two different dispatchers, this is similar.